Makefile

```diff
@@ -63,6 +63,8 @@ deploy:
 	cd config/manager && kustomize edit set image controller=${IMG}
 	kustomize build config/default | kubectl apply -f -
 
+protobuf:
+	protoc -I=api/proxy --go_out=plugins=grpc:api/proxy api/proxy/*.proto
```

Comment:
How is the `protoc` binary built/fetched? gRPC and protoc-gen-go are finicky about versioning and require the `protoc` and grpc client library version to be kept in sync

Reply:
It's up to the developer to manage that. We should probably include it in the development instructions but you only need them at generate-time. So not in CI.

Reply:
So you're also going to need a 'verify' script in CI to ensure that the committed generated source file is up to date. Alternatively you could look at using `go generate` but iirc that can be more pain than it is worth, especially at this point.

Typically when I write a 'verify' script, I write a corresponding 'update' script which is responsible for regenerating the files, and the verify script actually just calls the update script and performs a 'diff'. There's lots of ways to do it though 😅

Reply:
So I've added a dockerfile under `hack/` that will build a container image with specific `protoc` and Go plugin versions. We need this to work with the slightly older version of gRPC we're running because etcd (even on 3.4.3) hasn't upgraded yet.

As a side effect I've added a `make` target `protoc` to use that image to generate updated Protobuf/gRPC Go code, and updated the `verify` target to check that there are no diffs to running this tool.

I've also upgraded the contributing documentation to explain this.